### PR TITLE
3240 Support to plot 3D and RGB color image

### LINKED ISSUE #3240: plot_2d_or_3d_image with colour
I have my ground truth image with overlaid segmentation. The image is in grayscale and the segmentation is in colour, the two have been blended together and a 2d slice is correctly displayed with `plt.imshow`. I'd like to log this image to tensorboard, for which I use `plot_2d_or_3d_image`. Since the image is 3D, a gif is created, but the colour information is lost. I use `max_channels=3`, but this creates 3 independent images. Would it be possible to enhance the `plot_2d_or_3d_image` code to allow colours if num_channels=3?

partially related: https://github.com/Project-MONAI/MONAI/pull/3242.

matplotlib
image

tensorboard
image

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3240 .

### Description
This PR added support to plot images with `TensorBoardX`, and added support to plot RGB color image.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Test result for 3D RGB image:

https://user-images.githubusercontent.com/27533405/140055077-b5476bfd-5082-453c-b2ff-fc257f3cd7a9.mov

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

This PR is ready for review now, just a strange CI error in python 3.6 it can't import something..

Thanks.

File: monai/visualize/img2tensorboard.py [hunk 1]
```diff
@@ -206,7 +224,12 @@ def plot_2d_or_3d_image(
 
     if d.ndim >= 4:
         spatial = d.shape[-3:]
-        for j, d3 in enumerate(d.reshape([-1] + list(spatial))[:max_channels]):
+        d = d.reshape([-1] + list(spatial))
+        if d.shape[0] == 3 and max_channels == 3 and has_tensorboardx and isinstance(writer, SummaryWriterX):  # RGB
+            writer.add_video(tag, d[None], step, fps=max_frames, dataformats="NCHWT")
```
Comment (wyli) [inline monai/visualize/img2tensorboard.py:229]: not sure why `max_frames` is used as the fps parameter, is it a typo?

Comment (Nic-Ma) [reply monai/visualize/img2tensorboard.py:229]: Hi @wyli ,

I think the `max_frames` parameter of GIF is same as FPS of video if we plot the 3D image as video, not correct?

Thanks.

Comment (Nic-Ma) [reply monai/visualize/img2tensorboard.py:230]: I updated the PR for FPS in the latest commit.

Thanks.

**Comment by Nic-Ma** [comment]
Optimized the logic for multi-channel GIF images, the test result for 2 channels:

https://user-images.githubusercontent.com/27533405/140509342-dd56fe31-0381-4e75-b1af-9520d0262861.mov

Thanks.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Comment by Nic-Ma** [comment]
I tried to verify the TensorBoard plotting of `image`, `label` and `output` in tutorial:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/unet_segmentation_3d_ignite.ipynb
Everything seems OK.

Thanks.

File: requirements-dev.txt [hunk 2]
```diff
@@ -41,3 +41,5 @@ einops
 transformers
 mlflow
 matplotlib
+tensorboardX
+moviepy
```
Comment (wyli) [inline requirements-dev.txt:45]: is this required by the dev environment but not the user's?

Comment (Nic-Ma) [reply requirements-dev.txt:45]: Hi @wyli ,

If you want to visualize video with TensorBoardX, you need to install `moviepy`, others plot with TensorBoardX don't need it. So I just added in the dev for test.
Maybe we should also add it to the general install logic `pip install monai[tensorboardX, moviepy]`.

Thanks.

Comment (Nic-Ma) [reply requirements-dev.txt]: removed in the latest commit.

Thanks.

**Pull request merged**